### PR TITLE
Add TeamDownloadRequestStrategy

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* Add `TeamDownloadRequestStrategy` to fetch a `Team` when `needsToBeUpdatedFromBackend` is `true`.
* Add `MemberDownloadRequestStrategy` to fetch a teams members when `needsToRedownloadMembers` is set to `true` on that `Team`.
* This strategy does not fetch the list of teams during sync, a separate strategy to do this will be added in a separate PR.

### CODE REVIEW

File: Source/Transcoders/MemberDownloadRequestStrategy.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    public func update(_ object: ZMManagedObject!, with response: ZMTransportResponse!, downstreamSync: ZMObjectSync!) {
+        guard downstreamSync as? ZMDownstreamObjectSync == self.downstreamSync, let team = object as? Team else { return }
+        team.needsToRedownloadMembers = false
+        team.updateMembers(with: response)
+    }
+
+    public func delete(_ object: ZMManagedObject!, downstreamSync: ZMObjectSync!) {
+        guard downstreamSync as? ZMDownstreamObjectSync == self.downstreamSync, let team = object as? Team else { return }
```
Comment (daehn) [inline Source/Transcoders/MemberDownloadRequestStrategy.swift]: I need to double check if we should delete the local team here as well e.g. when receiving a `404` on the `/team/{id}/members` endpoint, but I think it makes sense.

Comment (daehn) [inline Source/Transcoders/MemberDownloadRequestStrategy.swift]: I will also add some tests for this case.

File: Source/Transcoders/MemberDownloadRequestStrategy.swift [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+}
+
+
+extension MemberDownloadRequestStrategy: ZMDownstreamTranscoder {
+
+    public func request(forFetching object: ZMManagedObject!, downstreamSync: ZMObjectSync!) -> ZMTransportRequest! {
+        guard downstreamSync as? ZMDownstreamObjectSync == self.downstreamSync, let team = object as? Team else { return nil }
+        return team.remoteIdentifier.map(TeamDownloadRequestFactory.getMembersRequest)
```
Comment (vytis) [inline Source/Transcoders/MemberDownloadRequestStrategy.swift]: Why not just simply call the method? Or it should be `flatMap` to unwrap optional?

File: Source/Transcoders/TeamDownloadRequestStrategy.swift [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        return [downstreamSync]
+    }
+
+}
+
+
+extension TeamDownloadRequestStrategy: ZMDownstreamTranscoder {
+
+    public func request(forFetching object: ZMManagedObject!, downstreamSync: ZMObjectSync!) -> ZMTransportRequest! {
+        guard downstreamSync as? ZMDownstreamObjectSync == self.downstreamSync, let team = object as? Team else { return nil }
```
Comment (vytis) [inline Source/Transcoders/TeamDownloadRequestStrategy.swift]: Maybe a `fatal()` would make it easier to debug

File: Source/Transcoders/MemberDownloadRequestStrategy.swift (hunk 2 above)
Comment (daehn) [inline Source/Transcoders/MemberDownloadRequestStrategy.swift]: It's `map` on `Optional` (`remoteIdentifier`) to unwrap the optional, apply the function and return the result or `nil`.

**Comment by daehn** [comment]
@vytis updated!

**Pull request merged**